### PR TITLE
Add Traveling Salesman Problem (TSP) Algorithms and Tests

### PR DESCRIPTION
**Pull request by ngtduc693**
### Summary
This PR adds a new module for solving the Traveling Salesman Problem (TSP) in C#. It includes two approaches:

**- Brute-force search:** Finds the optimal route by checking all possible permutations. Suitable for small numbers of cities.
**- Nearest neighbor heuristic:** Builds a route by always visiting the nearest unvisited city next. Fast but may not be optimal.

AND, 

- All code is written using modern C# 10+ syntax and best practices.
- Comments are detailed and clear for educational purposes.

### Details

Implements TravelingSalesmanSolver with detailed XML comments explaining the algorithms and usage.
Provides comprehensive unit tests covering all edge cases, invalid inputs, and correctness for both methods.
Updates the README to include TSP in the list of supported problems, with a description and links to the code.

### Usage

The brute-force method guarantees the shortest route for small city counts.
The nearest neighbor method is efficient for larger problems and provides a reasonable approximation.

### UT evidence

image

- [x] I have performed a self-review of my code
- [X] My code follows the style guidelines of this project
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Comments in areas I changed are up to date
- [X] I have added comments to hard-to-understand areas of my code
- [X] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Problems/TravelingSalesman/TravelingSalesmanSolver.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            for (int i = 0; i < arr.Length; i++)
+            {
+                var rest = arr.Where((_, idx) => idx != i).ToArray();
+                foreach (var perm in Permute(rest))
+                {
+                    yield return new[] { arr[i] }.Concat(perm).ToArray();
+                }
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (ngtduc693) [reply Algorithms/Problems/TravelingSalesman/TravelingSalesmanSolver.cs:135]: Out of necessity, I intentionally used newer and more advanced C# techniques

**Comment by ngtduc693** [comment]
> Please resolve conflicts

@siriak  I'm done, thanks
